### PR TITLE
Rename Google.Cloud.Firestore.Data to Google.Cloud.Firestore

### PR DESCRIPTION
**Pull request by jskeet**
(Lots of files, but the changes are trivial.)

### CODE REVIEW

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationTests/coverage.xml [hunk 1]
```diff
@@ -17,5 +17,5 @@
     <AttributeFilterEntry>System.Diagnostics.DebuggerNonUserCodeAttribute</AttributeFilterEntry>
   </AttributeFilters>
   <TargetWorkingDir>.</TargetWorkingDir>
-  <Output>../../../coverage/Google.Cloud.Firestore.Data.Snippets.dvcr</Output>
+  <Output>../../../coverage/Google.Cloud.Firestore.IntegrationTests.dvcr</Output>
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationTests/coverage.xml]: Is this correct? It's more than just removing ".Data"

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Snippets/coverage.xml [hunk 2]
```diff
@@ -17,5 +17,5 @@
     <AttributeFilterEntry>System.Diagnostics.DebuggerNonUserCodeAttribute</AttributeFilterEntry>
   </AttributeFilters>
   <TargetWorkingDir>.</TargetWorkingDir>
-  <Output>../../../coverage/Google.Cloud.Firestore.Data.Tests.dvcr</Output>
+  <Output>../../../coverage/Google.Cloud.Firestore.Snippets.dvcr</Output>
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Snippets/coverage.xml]: and here

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/coverage.xml [hunk 3]
```diff
@@ -17,5 +17,5 @@
     <AttributeFilterEntry>System.Diagnostics.DebuggerNonUserCodeAttribute</AttributeFilterEntry>
   </AttributeFilters>
   <TargetWorkingDir>.</TargetWorkingDir>
-  <Output>../../../coverage/Google.Cloud.Firestore.Data.IntegrationTests.dvcr</Output>
+  <Output>../../../coverage/Google.Cloud.Firestore.Tests.dvcr</Output>
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/coverage.xml]: and here

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationTests/coverage.xml (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationTests/coverage.xml]: It's git or github getting confused - it thinks it's a rename from ...Data.Snippets/coverage.xml to ...Data.IntegrationTests/coverage.xml.

**Pull request merged**